cmd/payload-testing/prow-plugin/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	configresolver "github.com/openshift/ci-tools/pkg/registry/server"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/githubeventserver"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+	"k8s.io/test-infra/prow/pjutil"
+)
+
+type options struct {
+	githubEventServerOptions githubeventserver.Options
+	github                   prowflagutil.GitHubOptions
+
+	webhookSecretFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+
+	o.github.AddFlags(fs)
+	o.githubEventServerOptions.Bind(fs)
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatalf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) Validate() error {
+	return o.githubEventServerOptions.DefaultAndValidate()
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	logger := logrus.WithField("plugin", "payload-testing")
+
+	o := gatherOptions()
+	if err := o.Validate(); err != nil {
+		logger.Fatalf("Invalid options: %v", err)
+	}
+
+	if err := secret.Add(o.github.TokenPath, o.webhookSecretFile); err != nil {
+		logger.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	getWebhookHMAC := secret.GetTokenGenerator(o.webhookSecretFile)
+
+	githubClient, err := o.github.GitHubClient(false)
+	if err != nil {
+		logger.WithError(err).Fatal("Error getting GitHub client.")
+	}
+	resolverClient := configresolver.NewResolverClient(api.URLForService(api.ServiceConfig))
+
+	serv := &server{
+		ghc:                githubClient,
+		ciopResolverClient: resolverClient,
+	}
+
+	eventServer := githubeventserver.New(o.githubEventServerOptions, getWebhookHMAC, logger)
+	eventServer.RegisterHandleIssueCommentEvent(serv.handleIssueComment)
+	eventServer.RegisterHelpProvider(helpProvider, logger)
+
+	interrupts.OnInterrupt(func() {
+		eventServer.GracefulShutdown()
```

Comment:
Not your code but why doesn't it just take a context and honor cancel?

Reply:
Well, this doesn't take wait for any context at all. `eventserver.Shutdown(ctx)` does, but this is being handled by interrupts. So we just need to pass a context to `interupts` instead.

Reply:
I think that what Steve means is that the eventServer should be changed to accept a context and honor cancel by gracefully shutting down. THen on interrupt you'd just cancel that context.

Reply:
Yes, this is not a blocker for now. The event server is already honoring a context in `Shutdown(ctx)`. The GracefullShutdown does something differently. It just waits for all the spawned goroutines to finish. Either way you can change this code to 
```go
interrupts.OnInterrupt(func() {
    eventServer.GracefulShutdown()
    if err := eventServer.Shutdown(myctx);err !=nil{.....}
}
```